### PR TITLE
324 classify method does not works when coming from cmd

### LINKED ISSUE #324: classify method does not works when coming from cmd+/
Trying to classify a method is not working, I don't know if always or just in some conditions.

### PR DESCRIPTION
**Pull request by estebanlm**
Fixes #324

### CODE REVIEW

File: src/Calypso-SystemTools-Core/ClyRequestMethodTagForEditorCommand.class.st [hunk 1]
```diff
@@ -7,6 +7,19 @@ Class {
...
+ClyRequestMethodTagForEditorCommand class >> canBeExecutedInCodeMethodEditor: aTool [
+	
+	^ true
+]
+
+{ #category : #testing }
+ClyRequestMethodTagForEditorCommand class >> canBeExecutedInContext: aToolContext [
+
+	^ (super canBeExecutedInContext: aToolContext)
+		and: [ aToolContext tool isKindOf: ClyMethodCodeEditorToolMorph ]
```
Comment (dionisiydk) [inline src/Calypso-SystemTools-Core/ClyRequestMethodTagForEditorCommand.class.st]: Old code here

File: src/Calypso-Browser/ClyBrowserToolMorph.class.st [hunk 2]
```diff
@@ -260,12 +260,6 @@ ClyBrowserToolMorph >> buildStatusBar [
 	self fillStatusBar
 ]
 
-{ #category : #controlling }
-ClyBrowserToolMorph >> cleanAfterRemove. [
-	self detachFromSystem.
-	decorators do: [ :each | each toolWasRemoved ]
-]
```
Comment (dionisiydk) [inline src/Calypso-Browser/ClyBrowserToolMorph.class.st]: Why it is removed?

File: src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st [hunk 3]
```diff
@@ -17,9 +17,22 @@ Class {
...
 ClyMethodEditorCommand class >> canBeExecutedInContext: aToolContext [
-	^aToolContext tool isKindOf: ClyMethodEditorToolMorph 
+
+	^ aToolContext tool isCommandAvailable: self
+]
+
+{ #category : #testing }
+ClyMethodEditorCommand class >> canBeExecutedInMethodEditor: aTool [
+
+	^ false
```
Comment (dionisiydk) [inline src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st]: It should be true by default

File: src/Calypso-SystemTools-Core/ClyMethodTagsAndPackageEditorMorph.class.st [hunk 4]
```diff
@@ -115,6 +115,12 @@ ClyMethodTagsAndPackageEditorMorph >> openEditor [
 	]
 ]
 
+{ #category : #accessing }
+ClyMethodTagsAndPackageEditorMorph >> ownerTool: anObject [
+
+	super ownerTool: anObject
```
Comment (dionisiydk) [inline src/Calypso-SystemTools-Core/ClyMethodTagsAndPackageEditorMorph.class.st]: Some garbage?

File: src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st (hunk 3 above)
Comment (estebanlm) [reply src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st]: no it shouldn't. If you put true by default you will have the same problem as before.

File: src/Calypso-SystemTools-Core/ClyMethodTagsAndPackageEditorMorph.class.st (hunk 4 above)
Comment (estebanlm) [reply src/Calypso-SystemTools-Core/ClyMethodTagsAndPackageEditorMorph.class.st]: yep, I put an enter while digging.

File: src/Calypso-SystemTools-Core/ClyRequestMethodTagForEditorCommand.class.st (hunk 1 above)
Comment (estebanlm) [reply src/Calypso-SystemTools-Core/ClyRequestMethodTagForEditorCommand.class.st]: oops. Yep :P

File: src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st (hunk 3 above)
Comment (dionisiydk) [reply src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st]: This method is related to main superclass of method editors ClyMethodEditorTool.
And all subclasses of ClyMethodEditorCommand are supposed to work with any such tools:
- you can edit tag and package in existing method editor and in new method editor.

Comment (dionisiydk) [reply src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st]: Maybe I not understand your intention. Do you want to disable command in "existing method" editor? (and keep it in method creation tool)

Comment (estebanlm) [reply src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st]: since it does not works, yes.

Comment (estebanlm) [reply src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st]: but when #defaultMenuItemName is called methodEditor is nil, so I do not have a way to know where it is going it act (as for change the name).

**Comment by dionisiydk** [comment]
Add method #readParametersInContext:.  it is called when command is
created. (prepareExecution.. Is called just before execute)

пт, 28 сент. 2018 г., 13:09 Esteban Lorenzano <notifications@github.com>:

File: src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st [hunk 5]
```diff
@@ -17,9 +17,22 @@ Class {
 	#category : #'Calypso-SystemTools-Core-Editors-Methods'
 }
 
+{ #category : #testing }
+ClyMethodEditorCommand class >> canBeExecutedInCodeMethodEditor: aTool [
```
Comment (dionisiydk) [inline src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st]: somehow this method is duplicated in file

File: src/Calypso-SystemTools-Core/ClyRequestMethodTagForEditorCommand.class.st [hunk 6]
```diff
@@ -25,3 +29,13 @@ ClyRequestMethodTagForEditorCommand >> execute [
 	tagEditor := methodEditor tagsAndPackageEditor.
 	tagEditor requestChangeBy: [ tagEditor requestTag] 
 ]
+
+{ #category : #'as yet unclassified' }
+ClyRequestMethodTagForEditorCommand >> readParametersFromContext: aToolContext [
+
+	menuItemName := aToolContext tool hasUnacceptedEdits
```
Comment (dionisiydk) [inline src/Calypso-SystemTools-Core/ClyRequestMethodTagForEditorCommand.class.st]: Idea was to refactor superclass  ClyMethodEditorCommand prepare method: 
```Smalltalk
ClyMethodEditorCommand>>prepareFullExecutionInContext: aToolContext
	super prepareFullExecutionInContext: aToolContext.
	
	methodEditor := aToolContext tool
```
It should be replaced by 
```Smalltalk
ClyMethodEditorCommand>>readParametersFromContext: aToolContext
	super readParametersFromContext: aToolContext.
	
	methodEditor := aToolContext tool
```
And then methodEditor can be used to implement conditional #defaultMenuItemName:
```Smalltalk
ClyRequestMethodTagForEditorCommand>>defaultMenuItemName
    methodEditor hasUnacceptedEdits
                ifTrue: [ 'Classify currently editing method' ]
 		ifFalse: [ 'Classify method' ]
```

File: .travis.yml [hunk 7]
```diff
@@ -3,9 +3,9 @@ sudo: false
 
 os: linux
 
-#smalltalk_edge:
-#  source: hpi-swa/smalltalkCI
-#  branch: dev
+smalltalk_edge:
+  source: estebanlm/smalltalkCI
```
[the remark this answers is not in the document]
Comment (estebanlm) [reply .travis.yml]: yes. is already in hpi-swa/smalltalkCI/dev but I will change it later.

File: scripts/preLoading.st [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
+Iceberg enableMetacelloIntegration: false.
 Metacello new
 	baseline: 'Calypso';
-	repository: 'gitlocal://src';
+	repository: 'tonel://src';
 	onConflict: [ :ex :loaded :incoming | ex useIncoming ];
 	onUpgrade: [ :ex :loaded :incoming | ex useIncoming ];
 	onWarningLog;
 	load.
+Iceberg enableMetacelloIntegration: true.
```
[the remark this answers is not in the document]
Comment (estebanlm) [reply scripts/preLoading.st]: yes

File: src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st (hunk 5 above)
Comment (estebanlm) [reply src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st]: no is not

**Pull request merged**